src/auth/jwt/jwt-auth.guard.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  ExecutionContext,
+  HttpException,
+  HttpStatus,
+  Injectable,
+} from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { request } from 'express';
+import { Observable } from 'rxjs';
+
+function AuthGuardMixin(strategy: string) {
+  abstract class MixinAuthGuard extends AuthGuard(strategy) {
+    canActivate(
+      context: ExecutionContext,
+    ): boolean | Promise<boolean> | Observable<boolean> {
+      const request = context.switchToHttp().getRequest();
+      const authorization = request.headers['authorization'];
+      if (!authorization) {
+        throw new HttpException('jwt must be provided', HttpStatus.BAD_REQUEST);
+      }
+      return super.canActivate(context);
+    }
+
+    handleRequest(
+      err: any,
+      user: any,
+      info: { message: string | Record<string, any> },
+      context: ExecutionContext,
+    ) {
+      try {
+        if (user) {
+          return super.handleRequest(err, user, info, context);
+        } else {
+          if (err instanceof HttpException) throw err;
+          else {
+            throw new HttpException(info.message, this.getStatus(info.message));
+          }
+        }
```

Comment:
```suggestion
        if (user) {
          return super.handleRequest(err, user, info, context);
        }
        
        if (err instanceof HttpException) throw err;
       
        throw new HttpException(info.message, this.getStatus(info.message));
```
이런 식으로 가는게 코드가 좀 더 깔끔할 것 같아요